src/ai/backend/test/testcases/session/create_single_node_multi_container_session.py

```diff
@@ -0,0 +1,63 @@
+import asyncio
+
+from ai.backend.client.session import AsyncSession
+from ai.backend.common.types import ClusterMode
+from ai.backend.test.testcases.session.context import ComputeSessionContext
+from ai.backend.test.testcases.template import TestCode
+
+# Test environment configuration
+# TODO: Make these configurable loaderable by template wrapper
+_IMAGE_NAME = "cr.backend.ai/stable/python:3.9-ubuntu20.04"
+_IMAGE_RESOURCES = {"cpu": 1, "mem": "512m"}
+_TEST_TIMEOUT = 10.0  # seconds
+
+
+# TODO: Refactor this using `parametrize` concept.
+class SingleNodeMultiContainerSessionCreation(TestCode):
+    async def test(self) -> None:
+        async with AsyncSession() as client_session:
+            session_name = ComputeSessionContext.get_current()
+
+            EXPECTED_EVENTS = {
+                "session_enqueued",
+                "session_scheduled",
+                "kernel_preparing",
+                "kernel_creating",
+                "kernel_started",
+                "session_started",
+            }
+
+            collected_events = set()
+
+            async def collect_events():
+                async with client_session.ComputeSession(session_name).listen_events() as events:
+                    async for event in events:
+                        collected_events.add(event.event)
+                        if collected_events == EXPECTED_EVENTS:
+                            # print("All expected events received.")
+                            break
+
+            listener_task = asyncio.create_task(
+                asyncio.wait_for(collect_events(), timeout=_TEST_TIMEOUT)
+            )
+
+            created_session = await client_session.ComputeSession.get_or_create(
+                _IMAGE_NAME,
+                name=session_name,
+                resources=_IMAGE_RESOURCES,
+                cluster_mode=ClusterMode.SINGLE_NODE,
+                cluster_size=3,
+            )
```

Comment:
Separately, but it seems that in this part, a session is created, and when the test ends, the session will not be cleaned up from backend.ai.

Reply:
Could you please try again?

The session spawned in the test must be cleaned up in the `SessionSetupTemplateWrapper`.